runners/s3-benchrunner-rust/src/lib.rs

```diff
@@ -0,0 +1,87 @@
+use serde::Deserialize;
+use std::{fs::File, io::BufReader, process};
+
+mod transfer_manager;
+pub use transfer_manager::TransferManagerRunner;
+
+fn exit_with_skip_code(msg: &str) -> ! {
+    eprintln!("Skipping benchmark - {msg}");
+    process::exit(123)
+}
+
+/// All configuration for a benchmark runner.
+/// Includes values from workload json file, and from the command line
+#[derive(Debug)]
+pub struct BenchmarkConfig {
+    pub workload: WorkloadConfig,
+    pub bucket: String,
+    pub region: String,
+    pub target_throughput_gigabits: f64,
+}
+
+/// From the workload's JSON file
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct WorkloadConfig {
+    pub version: i32,
+    pub files_on_disk: bool,
+    pub checksum: Option<String>,
```

Comment:
This should probably be something less ambiguous like `Bytes`. I always get confused otherwise.
```suggestion
    pub checksum: Option<String>,
```

Reply:
We could also use a hex encoding crate.

Reply:
It's the string "CRC32" or similar. It's the kind of checksums we'll use in the benchmark. This JSON file is described here: https://github.com/awslabs/aws-crt-s3-benchmarks/tree/main/workloads#readme

Reply:
I'll try turning it into an enum, and getting serde derive to work nicely with it